### PR TITLE
Stop populating legacy DNS names

### PR DESCRIPTION
**Pull request by thockin**
@ArtfulCoder @vishh I have not actually tested this one yet :)  If anyone has time to take this over and test the crap out of it, that would be great...

### CODE REVIEW

File: cluster/addons/dns/kube2sky/kube2sky.go [hunk 1]
```diff
@@ -215,41 +212,31 @@ func (ks *kube2sky) addDNSUsingEndpoints(subdomain string, e *kapi.Endpoints, is
...
 	}
-	return ks.generateRecordsForHeadlessService(subdomain, e, svc, isNewStyleFormat)
+	return ks.generateRecordsForHeadlessService(subdomain, e, svc)
 }
 
 func (ks *kube2sky) handleEndpointAdd(obj interface{}) {
 	if e, ok := obj.(*kapi.Endpoints); ok {
 		name := buildDNSNameString(ks.domain, e.Namespace, e.Name)
-		ks.mutateEtcdOrDie(func() error { return ks.addDNSUsingEndpoints(name, e, false) })
+		ks.mutateEtcdOrDie(func() error { return ks.addDNSUsingEndpoints(name, e) })
```
Comment (ArtfulCoder) [inline cluster/addons/dns/kube2sky/kube2sky.go]: the intent is to delete this line ?

**Comment by ArtfulCoder** [comment]
the dns e2e needs to be updated.
otherwise, this looks good already :)

**Comment by thockin** [comment]
Updated.  I am turning up a cluster right now.

**Comment by vishh** [comment]
LGTM. Thanks for the cleanup Tim!

**Comment by thockin** [comment]
I turned up a cluster with a new skydns built from this and it work.

**Comment by vishh** [comment]
Will merge once Shippable CI passes.

**Comment by ArtfulCoder** [comment]
@thockin somewhere in kubelet, shouldn't we remove the extraneous search domain now ?

**Comment by thockin** [comment]
That was removed months ago

**Comment by vishh** [comment]
kube2sky unit tests have to be updated

**Comment by thockin** [comment]
Good catch!

**Pull request merged**